File: ios/BiblosSurfer/Tools/AccessibilityIdentifiers.swift
//
//  AccessibilityIdentifiers.swift
//  BiblosSurfer
//
//  Created by Mikołaj Linczewski on 21/08/2026.
//

import Foundation

enum AccessibilityIdentifiers {
    enum Library {
        static let title = "library.title"
        static let importButton = "library.importButton"
        static let grid = "library.grid"
        static let emptyState = "library.emptyState"
        static let loading = "library.loading"

        static func cell(_ bookTitle: String) -> String {
            "library.cell.\(bookTitle)"
        }
    }

    enum Reader {
        static let container = "reader.container"
        static let title = "reader.title"
        static let close = "reader.close"
        static let settings = "reader.settings"
        static let progress = "reader.progress"
        static let ttsPlay = "reader.ttsPlay"
        static let ttsPause = "reader.ttsPause"
        static let ttsStop = "reader.ttsStop"
        static let ttsNext = "reader.ttsNext"
        static let ttsPrevious = "reader.ttsPrevious"
        static let ttsPanel = "reader.ttsPanel"
    }

    enum Settings {
        static let fontSize = "settings.fontSize"
        static let fontFamily = "settings.fontFamily"
        static let theme = "settings.theme"
        static let scrollMode = "settings.scrollMode"
        static let tts = "settings.tts"
        static let voice = "settings.voice"
        static let speechRate = "settings.speechRate"
        static let pitch = "settings.pitch"
        static let volume = "settings.volume"
        static let preUtteranceDelay = "settings.preUtteranceDelay"
        static let postUtteranceDelay = "settings.postUtteranceDelay"
        static let language = "settings.language"
        static let chunkUnit = "settings.chunkUnit"
        static let systemSpeech = "settings.systemSpeech"
        static let done = "settings.done"
    }

    enum Error {
        static let title = "error.title"
        static let description = "error.description"
        static let dismiss = "error.dismiss"
    }
}

File: android/app/src/main/java/miko/biblossurfer/util/AccessibilityIdentifiers.kt
package miko.biblossurfer.util

object AccessibilityIdentifiers {
    object Library {
        const val title = "library.title"
        const val importButton = "library.importButton"
        const val grid = "library.grid"
        const val emptyState = "library.emptyState"
        const val loading = "library.loading"

        fun cell(bookTitle: String): String = "library.cell.$bookTitle"
    }

    object Reader {
        const val container = "reader.container"
        const val title = "reader.title"
        const val close = "reader.close"
        const val settings = "reader.settings"
        const val progress = "reader.progress"
        const val ttsPlay = "reader.ttsPlay"
        const val ttsPause = "reader.ttsPause"
        const val ttsStop = "reader.ttsStop"
        const val ttsNext = "reader.ttsNext"
        const val ttsPrevious = "reader.ttsPrevious"
        const val ttsPanel = "reader.ttsPanel"
    }

    object Settings {
        const val fontSize = "settings.fontSize"
        const val fontFamily = "settings.fontFamily"
        const val theme = "settings.theme"
        const val scrollMode = "settings.scrollMode"
        const val tts = "settings.tts"
        const val voice = "settings.voice"
        const val speechRate = "settings.speechRate"
        const val pitch = "settings.pitch"
        const val volume = "settings.volume"
        const val preUtteranceDelay = "settings.preUtteranceDelay"
        const val postUtteranceDelay = "settings.postUtteranceDelay"
        const val language = "settings.language"
        const val chunkUnit = "settings.chunkUnit"
        const val systemSpeech = "settings.systemSpeech"
        const val done = "settings.done"
    }

    object Error {
        const val title = "error.title"
        const val description = "error.description"
        const val dismiss = "error.dismiss"
    }
}
